compiler/back_end/util/code_template_test.py

```diff
@@ -14,85 +14,85 @@
 
 """Tests for code_template."""
 
+import string
 import unittest
 from compiler.back_end.util import code_template
 
+def _format_template_str(template: str, **kwargs) -> str:
+  return code_template.format_template(string.Template(template), **kwargs)
 
 class FormatTest(unittest.TestCase):
   """Tests for code_template.format."""
 
   def test_no_replacement_fields(self):
-    self.assertEqual("foo", code_template.format_template("foo"))
-    self.assertEqual("{foo}", code_template.format_template("{foo}"))
-    self.assertEqual("$foo$", code_template.format_template("$foo$"))
-    self.assertEqual("$_foo$", code_template.format_template("$_foo$"))
-    self.assertEqual("$foo_$", code_template.format_template("$foo_$"))
+    self.assertEqual("foo", _format_template_str("foo"))
+    self.assertEqual("{foo}", _format_template_str("{foo}"))
 
   def test_one_replacement_field(self):
-    self.assertEqual("foo", code_template.format_template("$_bar_$", bar="foo"))
+    self.assertEqual("foo", _format_template_str("${bar}", bar="foo"))
     self.assertEqual("bazfoo",
-                     code_template.format_template("baz$_bar_$", bar="foo"))
+                     _format_template_str("baz${bar}", bar="foo"))
     self.assertEqual("foobaz",
-                     code_template.format_template("$_bar_$baz", bar="foo"))
+                     _format_template_str("${bar}baz", bar="foo"))
     self.assertEqual("bazfooqux",
-                     code_template.format_template("baz$_bar_$qux", bar="foo"))
+                     _format_template_str("baz${bar}qux", bar="foo"))
 
   def test_one_replacement_field_with_formatting(self):
-    self.assertEqual("1.000000",
-                     code_template.format_template("$_bar:.6f_$", bar=1))
-    self.assertEqual("'foo'",
-                     code_template.format_template("$_bar!r_$", bar="foo"))
-    self.assertEqual("==foo==",
-                     code_template.format_template("$_bar:=^7_$", bar="foo"))
-    self.assertEqual("=='foo'==",
-                     code_template.format_template("$_bar!r:=^9_$", bar="foo"))
-    self.assertEqual("xx=='foo'==yy",
-                     code_template.format_template("xx$_bar!r:=^9_$yy",
-                                                   bar="foo"))
+    # Basic string.Templates don't support formatting values.
```

Comment:
There have been a few different ways to do formatting in Python so maybe I'm mistaken but I thought the tests from line 43-51 should still work.

Reply:
They do not (the following line shows that a `ValueError` is raised to make that clear).